### PR TITLE
search: implement GQL query printers

### PR DESCRIPTION
**Pull request by rvantonder**
Implements printers (follow up of https://github.com/sourcegraph/sourcegraph/pull/41835)

## Test plan
Exhaustively tested via GQL console. This is a debugging/observability utility and underlying functionality is already tested.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/parse_search_query.go [hunk 1]
```diff
@@ -25,14 +53,83 @@ func (r *schemaResolver) ParseSearchQuery(ctx context.Context, args *struct {
...
+		jsonString := printer.JSONVerbose(j, verbosity)
+		return &JSONValue{Value: jsonString}, nil
+	case Sexp:
+		sexpString := printer.SexpVerbose(j, verbosity, true)
+		return &JSONValue{Value: sexpString}, nil
+	case Mermaid:
+		if args.OutputVerbosity != Minimal {
+			return nil, errors.New("unsupported output options for MERMAID JOB_TREE, only MINIMAL verbosity is supported")
+		}
+		mermaidString := printer.Mermaid(j)
```
Comment (rvantonder) [inline cmd/frontend/graphqlbackend/parse_search_query.go:131]: I'd like mermaid output to honor verbosity but currently only `VerbosityNone` outputs a valid mermaid diagram. Here's example output for `VerbosityBasic` that doesn't parse:

```
flowchart TB
0[ALERT <br> query:  <br> originalQuery:  <br> patternType: literal]
  0---1
  1[FEELINGLUCKYSEARCH]
    1---2
    2[TIMEOUT <br> timeout: 20s]
      2---3
      3[LIMIT <br> limit: 500]
        3---4
        4[PARALLEL]
          4---5
          5[REPOPAGER <br> repoOpts.repoFilters.0: sourcegraph]
            5---6
            6[PARTIALREPOS]
              6---7
              7[ZOEKTREPOSUBSETTEXTSEARCH <br> query: (and substr:"func /.*parse.*/" file_substr:".go") <br> type: text]
                4---8
          8[REPOSCOMPUTEEXCLUDED <br> repoOpts.repoFilters.0: sourcegraph]
            4---9
          9[REPOPAGER <br> repoOpts.repoFilters.0: sourcegraph]
            9---10
            10[PARTIALREPOS]
              10---11
              11[SEARCHERTEXTSEARCH <br> indexed: false]
```

Error:


Screen Shot 2022-09-21 at 5 08 52 PM

Maybe @camdencheek has an idea to make this valid, I didn't look deeply into it

Comment (camdencheek) [reply cmd/frontend/graphqlbackend/parse_search_query.go:131]: This seems to be an issue with special characters in the query. [Escaping the parens and the quotes](https://mermaid.live/edit#pako:eNqlk11vmzAUhv-K5Um9qBoKadI2aJpEE7dDJSPjQ-oG0-SCk1gDTI3RGlX97zt8tCWptIvtiuPXj1_j99hPOBEpwyZeZ-J3sqVSoeAqLvTIcogXoI_38hN6qJncmagbCMk3vKDZ16FYUqWYLIJdyUyUcahp9iMuENJHo5HRFEZ0TYhjf7lxwvntN59Y3vxzS8AUMOOuHEeBvSRu2G-seM5ErUw01qseBgbws5fBWeTYS7vHM55zgKe6_goDAfjkbTiJVpZnOQ5xBgzIQE2HwjTyyMpdWTfE68wlK4VbqkprimuewRkrTTdRJWqZsI2k5XbPESzA83xfOm92D2zLacz9Ax6mYcXFoXgRfXfJbdCuCK98EgTkLugC3GvP0YeJTosUVfV9paR59FALta6LBJ1qxyWVFdOOT1sRrXnGfg4xbSPaL1gYffJtJxV7VO9-sgvrcihftmH5c3e5CgNC7uZOuCCLf82tsZ8Npdn_92LW3EN9XzP0v3fDaC-v8U42oi59eB8HneBFyh5ZaqI1zSoGhvgE50zmlKfwwp4apxirLctZjE0oUyp_xTgunoGry5QqRlKuhMRma3CCaa2EvysSbCpZsxdowSmcMe-p5z8aAA5X) seems to do the trick. I don't know whether I'll be able to get to this immediately, but it should be a pretty easy fix

Comment (camdencheek) [reply cmd/frontend/graphqlbackend/parse_search_query.go:131]: Fix [here](https://github.com/sourcegraph/sourcegraph/pull/41934)

File: cmd/frontend/graphqlbackend/parse_search_query.go [hunk 2]
```diff
@@ -3,16 +3,44 @@ package graphqlbackend
...
+type args struct {
 	Query           string
 	PatternType     string
 	OutputPhase     string
 	OutputFormat    string
 	OutputVerbosity string
-}) (*JSONValue, error) {
+}
+
+func (r *schemaResolver) ParseSearchQuery(ctx context.Context, args *args) (*JSONValue, error) {
```
Comment (camdencheek) [inline cmd/frontend/graphqlbackend/parse_search_query.go:43]: Why do we return a `JSONValue` here rather than a string?

**Review by camdencheek — APPROVED** [review]

File: cmd/frontend/graphqlbackend/parse_search_query.go (hunk 2 above)
Comment (rvantonder) [reply cmd/frontend/graphqlbackend/parse_search_query.go:43]: Originally this result was only ever a JSON object. I'll simplify in follow up

**Pull request merged**